pkg/costmodel/clusters/clustermap.go

```diff
@@ -391,3 +355,63 @@ func (pcm *PrometheusClusterMap) StopRefresh() {
 		pcm.stop = nil
 	}
 }
+
+// MapToClusterInfo returns a ClusterInfo using parsed data from a string map. If
+// parsing the map fails for id and/or name, an error is returned.
+func MapToClusterInfo(info map[string]string) (*ClusterInfo, error) {
+	var id string
+	var name string
+
+	if i, ok := info[ClusterInfoIdKey]; ok {
+		id = i
+	} else {
+		return nil, fmt.Errorf("Cluster Info Missing ID")
+	}
+	if n, ok := info[ClusterInfoNameKey]; ok {
+		name = n
+	} else {
+		name = id
+	}
```

Comment:
This case used to be an error, what's the reason for the change?

Reply:
Poor logic. If we have the id, the name should always inherit the id as the name. Additionally, as we expand the definition of ClusterInfo beyond the scope of just prom, being more precise here is important. Name is not a required cluster info configuration, therefore, we shouldn't error when it's missing.

Reply:
Makes perfect sense to me!